tools/Environments/DevHome.Environments/Helpers/DataExtractor.cs

```diff
@@ -32,9 +33,46 @@ public static List<OperationsViewModel> FillDotButtonOperations(ComputeSystem co
             operations.Add(new OperationsViewModel("Delete", "\uE74D", computeSystem.DeleteAsync));
         }
 
-        // ToDo: Correct the function used
-        // operations.Add(new OperationsViewModel("Pin To Taskbar", "\uE718", computeSystem.));
-        // operations.Add(new OperationsViewModel("Add to Start Menu", "\uE8A9", computeSystem.));
+        return operations;
+    }
+
+    public static async Task<List<OperationsViewModel>> FillDotButtonPinOperationsAsync(ComputeSystem computeSystem)
+    {
+        var supportedOperations = computeSystem.SupportedOperations;
+        var operations = new List<OperationsViewModel>();
+        if (supportedOperations.HasFlag(ComputeSystemOperations.PinToTaskbar))
+        {
+            var pinResultTaskbar = await computeSystem.GetIsPinnedToTaskbarAsync();
+            if (pinResultTaskbar.Result.Status == ProviderOperationStatus.Success)
+            {
+                if (pinResultTaskbar.IsPinned)
+                {
+                    operations.Add(new OperationsViewModel("Unpin From Taskbar", "\uE77A", computeSystem.UnpinFromTaskbarAsync));

```

Comment:
Should this ViewModel be DI'ed?

Reply:
Sorry, what is DI'ed?

Reply:
Dependency Injection. Added as a 'service'

Reply:
Got it.  Changing whether or not these OperationsViewModel should come from a service is beyond the scope of this PR.  Please feel free to start up an email thread if you are interested.